src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/BaseDiagnosticAndGeneratorItemSource.cs

```diff
@@ -7,187 +7,209 @@
 using System.Collections.Immutable;
```

Comment:
this file is the meat of the change.  It moved from synchronously retrieving and populating hte 'items' value within, to asynchronously populating and updating it using a queue.